File: frontend/src/types/locator.ts
export interface Locator {
  id: number
  name: string
  description: string | null
  user_id: number
  created_at: string
  updated_at: string
  sublocators?: Sublocator[]
}

export interface LocatorCreate {
  name: string
  description?: string
}

export interface LocatorUpdate {
  name?: string
  description?: string
}

export interface Sublocator {
  id: number
  name: string
  description: string | null
  locator_id: number
  created_at: string
  updated_at: string
}

export interface SublocatorCreate {
  name: string
  description?: string
}

export interface SublocatorUpdate {
  name?: string
  description?: string
}

File: backend/app/schemas/locator.py
from datetime import datetime
from pydantic import BaseModel, ConfigDict


class LocatorBase(BaseModel):
    name: str
    description: str | None = None


class LocatorCreate(LocatorBase):
    pass


class LocatorUpdate(BaseModel):
    name: str | None = None
    description: str | None = None


class LocatorResponse(BaseModel):
    id: int
    name: str
    description: str | None
    user_id: int
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


class SublocatorBase(BaseModel):
    name: str
    description: str | None = None


class SublocatorCreate(SublocatorBase):
    pass


class SublocatorUpdate(BaseModel):
    name: str | None = None
    description: str | None = None


class SublocatorResponse(BaseModel):
    id: int
    name: str
    description: str | None
    locator_id: int
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)
